compiler/rustc_ast/src/ast.rs

```diff
@@ -1230,6 +1235,8 @@ pub enum ExprKind {
     Box(P<Expr>),
     /// An array (`[a, b, c, d]`)
     Array(Vec<P<Expr>>),
+    /// AnonConst
+    AnonConst(AnonConst),
```

Comment:
```suggestion
    ConstBlock(AnonConst),
```
(For patterns as well.)